### DISCUSSION TITLE
Which Key seems to affect keymap

### DISCUSSION THREAD

**Discussion by iamxiaojianzheng** [LazyVim/LazyVim#4016] [Q&A] [answered]
This is my custom keymap, and it cannot be used normally after updating to the latest lazyvim.

```lua
local map = LazyVim.safe_keymap_set
map({ "n", "o" }, "dd", '"_dd', { desc = "Delete Line" })
map({ "n", "v" }, "<C-x>", "dd", { desc = "Cut Line" })
```

When I press a `d`, the following message appears.

![image](https://github.com/user-attachments/assets/38935e62-3453-4756-b2d6-21168d1e9de9)

When I press `dd` or `<C-x>`, the following message appears.

![image](https://github.com/user-attachments/assets/aa16e493-e7e7-484d-8cd7-6dedf805179b)

This is not the expected effect, and normally the current line should be deleted or cut.

**Comment by folke** [comment]
Can you open an issue in the which-key repo? tx

**Reply by iamxiaojianzheng** [reply]
https://github.com/folke/which-key.nvim/issues/648

**Reply by iamxiaojianzheng** [reply, marked as answer]
https://github.com/folke/which-key.nvim/pull/642

**Comment by folke** [comment]
Technically, your mapping is actually wrong and that's what giving issues in which-key.

You're adding an op-pending keymap called `dd`.
So to execute it you'd have to do `ddd`.

What you probably want to do is map `dd` in normal mode and `d` in op-pending mode.

**Reply by iamxiaojianzheng** [reply]
Ok, but this configuration worked fine until I updated lazyvim. /(ㄒoㄒ)/~~

**Reply by iamxiaojianzheng** [reply]
Could you teach me how to configure to achieve this function?

**Comment by erlangparasu** [comment]
which-key is not shown up when typing `<space>` in toggleterm window

**Reply by folke** [reply]
Do you have any `<leader>` keymaps set in terminal mode?
If not , then they won't be shown of course...

**Reply by folke** [reply]
hmm, although, normal mode mappings probably also should work in terminal mode. Need to verify

**Reply by erlangparasu** [reply]
it works yesterday. stopped working after update lazyvim today.
when focusing on a file window, `space` works in normal mode. but when in a toggleterm window, it doesn't work, even in normal mode.

**Reply by erlangparasu** [reply]
<img width="866" alt="image" src="https://github.com/user-attachments/assets/93965059-9e1e-41d9-aca2-eca0842ab1ec">

**Reply by erlangparasu** [reply]
> hmm, although, normal mode mappings probably also should work in terminal mode. Need to verify

tried using `:terminal` to open new terminal buffer. in normal mode, `space` also not working to open which-key popup

**Reply by erlangparasu** [reply]
This is my config `lazy-lock.json` :
https://github.com/erlangparasu/dot-config-nvim/blob/da9214e90591715980e02a507759a6e73feafe87/lazy-lock.json

**Reply by folke** [reply]
should work now. I incorrectly mapped `nt` mode to `t` instead of `n`
